src/components/DatePicker.tsx

```diff
@@ -1,67 +1,44 @@
-import { createStyles, makeStyles } from '@material-ui/core';
-import { KeyboardDatePicker } from '@material-ui/pickers';
-import { MaterialUiPickersDate } from '@material-ui/pickers/typings/date';
 import moment from 'moment';
 import 'moment/min/locales';
 import React, { KeyboardEventHandler } from 'react';
-
-const useStyles = makeStyles(() =>
-  createStyles({
-    root: {
-      margin: 0,
-    },
-  })
-);
+import { TextField } from '@mui/material';
+import { LocalizationProvider, DesktopDatePicker } from '@mui/x-date-pickers';
+import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns';
 
 export interface Props {
   id: string;
   label: React.ReactNode;
   value?: string | null;
-  onChange: (id: string, value?: string) => void;
+  onChange: (id: string, value?: string | null) => void;
   'aria-label': string;
   onKeyPress?: KeyboardEventHandler;
   maxDate?: string;
   error?: boolean;
   helperText?: string;
   disabled?: boolean;
+  className?: string;
+  autocomplete?: string;
+  autoOk?: boolean;
 }
 
 export default function DatePicker(props: Props): JSX.Element {
-  const classes = useStyles();
-  const onDateChange = (date: MaterialUiPickersDate) => {
-    if (date && date.isValid()) {
-      props.onChange(props.id, date?.toISOString());
-    } else {
-      props.onChange(props.id, date?.toString());
-    }
-  };
-
   React.useEffect(() => {
     moment.locale([window.navigator.language, 'en']);
-  }, []);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [window.navigator.language]);
 
   return (
-    <KeyboardDatePicker
-      id={props.id}
-      value={props.value ?? null}
-      onChange={onDateChange}
-      label={props.label}
-      KeyboardButtonProps={{
-        'aria-label': props['aria-label'],
-      }}
-      disableToolbar={true}
-      variant='inline'
-      inputVariant='outlined'
-      format={moment.localeData().longDateFormat('L')}
-      margin='normal'
-      size='small'
-      fullWidth={true}
-      className={classes.root}
-      onKeyPress={props.onKeyPress}
-      maxDate={props.maxDate ?? undefined}
-      error={props.error}
-      helperText={props.helperText}
-      disabled={props.disabled}
-    />
+    <LocalizationProvider dateAdapter={AdapterDateFns}>
```

Comment:
@constanzauanini I introduced this change as mui doc show this https://mui.com/x/react-date-pickers/date-picker/ so we don't have all clients introducting localization code.

Reply:
Ok! In terraware-web we should delete all the LocalizationProviders that I added